src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java

```diff
@@ -0,0 +1,250 @@
+    protected int assemblyRegionPadding = defaultAssemblyRegionPadding();
+
+    @Advanced
+    @Argument(fullName = "activeProbabilityThreshold", shortName = "activeProbabilityThreshold", doc="Minimum probability for a locus to be considered active.", optional = true)
+    protected double activeProbThreshold = defaultActiveProbThreshold();
+
+    @Advanced
+    @Argument(fullName = "maxProbPropagationDistance", shortName = "maxProbPropagationDistance", doc="Upper limit on how many bases away probability mass can be moved around when calculating the boundaries between active and inactive assembly regions", optional = true)
+    protected int maxProbPropagationDistance = defaultMaxProbPropagationDistance();
+
+    @Argument(fullName = "disable_all_read_filters", shortName = "f", doc = "Disable all read filters", common = false, optional = true)
+    public boolean disableAllReadFilters = false;
+
+    /**
+     * @return Default value for the {@link #readShardSize} parameter, if none is provided on the command line
+     */
+    protected abstract int defaultReadShardSize();
+
+    /**
+     * @return Default value for the {@link #readShardPadding} parameter, if none is provided on the command line
+     */
+    protected abstract int defaultReadShardPadding();
+
+    /**
+     * @return Default value for the {@link #minAssemblyRegionSize} parameter, if none is provided on the command line
+     */
+    protected abstract int defaultMinAssemblyRegionSize();
+
+    /**
+     * @return Default value for the {@link #maxAssemblyRegionSize} parameter, if none is provided on the command line
+     */
+    protected abstract int defaultMaxAssemblyRegionSize();
+
+    /**
+     * @return Default value for the {@link #assemblyRegionPadding} parameter, if none is provided on the command line
+     */
+    protected abstract int defaultAssemblyRegionPadding();
+
+    /**
+     * @return Default value for the {@link #activeProbThreshold} parameter, if none is provided on the command line
+     */
+    protected abstract double defaultActiveProbThreshold();
+
+    /**
+     * @return Default value for the {@link #maxProbPropagationDistance} parameter, if none is provided on the command line
+     */
+    protected abstract int defaultMaxProbPropagationDistance();
+
+    @Override
+    public final boolean requiresReads() { return true; }
+
+    @Override
+    public final boolean requiresReference() { return true; }
+
+    private List<ReadShard> readShards;
+    private ReadShard currentReadShard;
+
+    /**
+     * Initialize data sources for traversal.
+     *
+     * Marked final so that tool authors don't override it. Tool authors should override onTraversalStart() instead.
+     */
+    @Override
+    protected final void onStartup() {
+        super.onStartup();
+
+        if ( readShardSize <= 0 ) {
+            throw new UserException.BadArgumentValue("--readShardSize", Integer.toString(readShardSize), "read shard size must be > 0");
```

Comment:
Declaring constants for all full/short names for all arguments made the argument list hard to read/understand (an effect that would be much worse for the `HaplotypeCaller` arguments themselves). I experimented with a reflection-based solution, but it was a bit silly. Decided ultimately to modify the error messages to describe the problem rather than including the exact argument names.